src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/ArgumentIsRequest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentValueResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\Argument\ArgumentMetadata;
+
+/**
+ * Supports the same instance as the request object passed along.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class ArgumentIsRequest implements ArgumentValueResolverInterface
```

Comment:
Finding the resolver is no problem, my thoughts went out to people who want to type `fooAction(Req| $request)` and find the suggestions mentioned:

![](http://i.imgur.com/L0LHsCy.png)